Mark TRANS_TTL_{01,02}, STAKE_POOLS_JOIN_05, and STAKE_POOLS_SMASH_01 pending

Anviking opened the pull request:
# Issue Number

None. Addressing CI failures.

# Overview


- [x] Add new `flakyBecauseOf ticketOrReason` helper that calls `pendingWith` unless `RUN_FLAKY_TESTS` is set.
- [x] Mark TRANS_TTL_{01,02} and STAKE_POOLS_JOIN_05 pending/flaky.
- [x] Also mark STAKE_POOLS_SMASH_01 pending
- [x] Add manual test calling for running flaky tests

# Comments

- Should lower the failure rate by 21% of runs, from 59% to 38%. 
- Next candidate for marking pending would be #2224, but with a relatively low failure rate of 3.6%, and being important, I think it would be a bad idea.
- Maybe we should have flaky tests run per default, unless setting `DONT_RUN_FLAKY_TESTS` in CI, to maximise the times we run them locally.

Recent bors failures:
```
succeded: 19, failed: 37 (66%), total: 56
excluding #expected failures

Broken down by tags/issues:
10 times #2292 Flaky test - various DB properties causing timeout | https://github.com/input-output-hk/cardano-wallet/issues/2292
7 times #2295 Flaky TRANS_TTL_{01,02} - SlotNo 80 > SlotNo 50 | https://github.com/input-output-hk/cardano-wallet/issues/2295
6 times
3 times #2311 Flaky test - integration test timeout after/related to STAKE_POOLS_LIST_01 | https://github.com/input-output-hk/cardano-wallet/issues/2311
3 times #2230 Flaky  STAKE_POOLS_JOIN_05 - Can join when stake key already exists | https://github.com/input-output-hk/cardano-wallet/issues/2230
2 times #2224 Flaky STAKE_POOLS_LIST_01 - List stake pools, has non-zero saturation & stake | https://github.com/input-output-hk/cardano-wallet/issues/2224
1 times #another-integration-timeout  |
1 times #2337 STAKE_POOLS_GARBAGE_COLLECTION_01 timed out | https://github.com/input-output-hk/cardano-wallet/issues/2337
1 times #2320 Flaky test - The node backend is unreachable at the moment. STAKE_POOLS_QUIT_02 | https://github.com/input-output-hk/cardano-wallet/issues/2320
1 times #2295, #2331 Flaky TRANS_TTL_{01,02} - SlotNo 80 > SlotNo 50 | https://github.com/input-output-hk/cardano-wallet/issues/2295
1 times #2207 Flaky SHELLEY_MIGRATE_01_big_wallet | https://github.com/input-output-hk/cardano-wallet/issues/2207
1 times #2118 Property `prop_rebalanceSelection` occasionally fails. | https://github.com/input-output-hk/cardano-wallet/issues/2118
```


<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

Anviking committed:
Add flakyBecauseOf hspec helper

like pendingWith, but disabled by setting
export RUN_FLAKY_TESTS=1

This can be done before release.

Anviking committed:
Mark TRANS_TTL_{01,02} and STAKE_POOLS_JOIN_05 flaky

Anviking committed:
Add FlakyTests.md manual test

piotr-iohk commented on test/manual/cardano-node/FlakyTests.md:
````diff
@@ -0,0 +1,9 @@
+Run them locally using 
+
+```bash
+RUN_FLAKY_TESTS=1 stack test cardano-wallet:integration
````
:+1:

piotr-iohk replied:
:thinking: would this work also with a windows test bundle? I guess it should, right?
```
set RUN_FLAKY_TESTS=1
cardano-wallet-test-integration.exe
```

Maybe could be added to the instruction.

piotr-iohk replied:
It would be also awesome to be able to run only flaky ones, but that's good enough I suppose :)

Anviking replied:
> Maybe could be added to the instruction.

Added 👍 

> It would be also awesome to be able to run only flaky ones

If we convert `flakyBecauseOf` to a wrapper around `it` rather than a wrapper around `pendingWith` we can customise the title of the test, making it easy to use `--match`. Wasn't sure how needed that was though. We can do later.

With -j 8, integration tests should be pretty fast anyway.

piotr-iohk reviewed (approved):
lgtm

Anviking committed:
Mark STAKE_POOLS_SMASH_01 pending

-j 8 was enabled on 2020-11-04. There _is_ a timeout two days after. But
timeouts only became _frequent_ on 2020-11-17.

 * [2020-11-06]|https://github.com/input-output-hk/cardano-wallet/pull/2297#issuecomment-723297161]
 * [2020-11-17|https://github.com/input-output-hk/cardano-wallet/pull/2319#issuecomment-729011351]
 * [2020-11-18|https://github.com/input-output-hk/cardano-wallet/pull/2329#issuecomment-729797752]
 * [2020-11-18|https://github.com/input-output-hk/cardano-wallet/pull/2329#issuecomment-730015347]
 * [2020-11-19|https://github.com/input-output-hk/cardano-wallet/pull/2335#issuecomment-730445221]
 * [2020-11-19|https://github.com/input-output-hk/cardano-wallet/pull/2335#issuecomment-730383980]
 * [2020-11-23|https://github.com/input-output-hk/cardano-wallet/pull/2338#issuecomment-732215930]
 * [2020-11-23|https://github.com/input-output-hk/cardano-wallet/pull/2338#issuecomment-732377904]
 * [2020-11-23|https://github.com/input-output-hk/cardano-wallet/pull/2338#issuecomment-732484867]
 * [2020-11-24|https://github.com/input-output-hk/cardano-wallet/pull/2338#issuecomment-732818214]
 * [2020-11-24|https://github.com/input-output-hk/cardano-wallet/pull/2343#issuecomment-733092684]
